modules/prebidServerBidAdapter/index.js

```diff
@@ -563,13 +563,15 @@ Object.assign(ORTB2.prototype, {
       const nativeParams = adUnit.nativeParams;
       let nativeAssets;
       if (nativeParams) {
+        let idCounter = 0;
         try {
           nativeAssets = nativeAssetCache[impressionId] = Object.keys(nativeParams).reduce((assets, type) => {
             let params = nativeParams[type];
 
             function newAsset(obj) {
               return Object.assign({
-                required: params.required ? 1 : 0
+                required: params.required ? 1 : 0,
+                id: (isNumber(params.id)) ? params.id : idCounter++
```

Comment:
@jsnellbaker - this doesn't cover the case where some assets have IDs and others don't. So there could wind up being duplicate IDs. I guess it might be ok to just document this behavior clearly.

Reply:
I agree it should be clearly documented, but I thought of a potential alternative that maybe could better handle the `id`s.  I'll push the commit shortly, let me know if it's worthwhile.